meirborowski/ts-grib2-toolkit
# Setup

### Setup TypeScript Project and Directory Structure

**Description:**  
This PR initializes the foundation for our `ts-grib2-toolkit`. It outlines the setup of a new TypeScript project, an organized directory structure, npm scripts, and the integration of ESLint and Prettier for consistent code quality.

**Changes:**  
- Initialized TypeScript with `tsc --init`.
- Created directories: `src`, `tests`, `types`, and `utils`.
- Defined npm scripts in `package.json` for building, testing, linting, and formatting.
- Integrated ESLint and Prettier with respective configuration files.
- Added a `README.md` with setup instructions.

**How to Test:**  
1. Clone the branch.
2. Run `npm install`.
3. Execute `npm run build` to ensure TypeScript transpiles to JavaScript.
4. Lint the project with `npm run lint` to check for any issues.
5. Use `npm run format` to format the codebase.

**Additional Notes:**  
For future iterations, tools like `ts-node` may be considered for a smoother development experience. The directory structure has been set to be modular to cater for future expansions.
